### DISCUSSION TITLE
2-way data binding in Nuxt3

### DISCUSSION THREAD

**Discussion by jamesolrog** [nuxt/nuxt#16996] [Questions] [answered]
I'm struggling to get 2-way data binding from Vue 3 ([see updates here](https://v3-migration.vuejs.org/breaking-changes/v-model.html#_3-x-syntax)) to work in my Nuxt 3 Application. Am I missing something?

Here is my code:

**Child Component**
```
// Components/Inputs/EmailField.vue

<template>
    <div>
        <label for="email">Email</label>
	    <input
		  type="email"
		  name="email"
		  id="email"
		  placeholder="you@example.com"
		  v-model="email"
            />
    </div>
</template>

<script setup>
    const props = defineProps({
	email: {
	    type: String,
	}
    });

    const emit = defineEmits(["update:email", "updateEmail"]);

    function updateEmail(email) {
	this.$emit("update:email", email);
    }
</script>
```

**Use in Parent:**
```
// Components/Form.vue

<template>
    <div>
	<form @submit.prevent="onSubmitClicked()">
		<InputsEmailField v-model:email="email" />
		<button type="submit">Submit</button>
	</form>
    </div>
</template>

<script setup>
    const email = "";
    // Have also tried: 'const email = ref("");'

    const onSubmitClicked = async () => {
	console.log(email);
    };
</script>
```

When the submit button is clicked, I expect to see the value of the email input being printed to console; however, it is always empty! This suggests that the update callback isn't working/firing? I wasn't sure if this could be a bug so wanted to seek some help first!

Thanks in advance.

**Comment by TheAlexLichter** [comment, marked as answer]
There are a few mistakes in your code. See [this StackBlitz](https://stackblitz.com/edit/nuxtframeworkdiscussions7009?file=components%2FEmailField.vue,components%2FForm.vue,app.vue) for a working version.

1. You should use a `ref` in the `Form` component (as commented out)
2. `this.$emit` is not available in `<script setup>`. Use `emit` instead which you've declared before
3. `v-model` on `<input>` can be fine but instead I'd opt in for manual binding as shown. If you want to use `v-model`, you need a watcher too.

**Reply by jamesolrog** [reply]
Ahhh thank you! Works like a dream. :)
